IO/BeagleBone/CANBusBBB.cs

```diff
@@ -1,169 +1,301 @@
+                }
+            }
+        }
+
+        /// <summary> Blocks the current thread and reads a CAN frame, returning the payload and the ID of the received CAN frame. </summary>
+        /// <returns> A tuple, with the first element being the ID of the received CAN frame and the second being the payload </returns>
+        public Tuple<uint, byte[]> Read()
+        {
+            if (Extended)
+            {
+                ExtendedCANFrame Frame = new ExtendedCANFrame();
+                read(Socket, ref Frame, Marshal.SizeOf(Frame));
+                byte[] Payload = new byte[Frame.DataLength];
+                unsafe
+                {
+                    for (int i = 0; i < Frame.DataLength; i++) { Payload[i] = Frame.Data[i]; }
+                }
+                return new Tuple<uint, byte[]>(Frame.CANID, Payload);
+            }
+            else
+            {
+                CANFrame Frame = new CANFrame();
+                read(Socket, ref Frame, Marshal.SizeOf(Frame));
+                byte[] Payload = new byte[Frame.DataLength];
+                unsafe
+                {
+                    for (int i = 0; i < Frame.DataLength; i++) { Payload[i] = Frame.Data[i]; }
+                }
+                return new Tuple<uint, byte[]>(Frame.CANID, Payload);
+            }
+        }
+
+        /// <summary> Reads a CAN frame asynchronously, returning the payload and the ID of the received CAN frame. </summary>
+        /// <returns> A task that will return a tuple, with the first element being the ID of the received CAN frame and the second being the payload </returns>
+        public Task<Tuple<uint, byte[]>> ReadAsync()
+        {
+            return Task.Run(() =>
+            {
+                if (Extended)
+                {
+                    ExtendedCANFrame Frame = new ExtendedCANFrame();
+                    read(Socket, ref Frame, Marshal.SizeOf(Frame));
+                    byte[] Payload = new byte[Frame.DataLength];
+                    unsafe
+                    {
+                        for (int i = 0; i < Frame.DataLength; i++) { Payload[i] = Frame.Data[i]; }
+                    }
+                    return new Tuple<uint, byte[]>(Frame.CANID, Payload);
+                }
+                else
+                {
+                    CANFrame Frame = new CANFrame();
+                    read(Socket, ref Frame, Marshal.SizeOf(Frame));
+                    byte[] Payload = new byte[Frame.DataLength];
+                    unsafe
+                    {
+                        for (int i = 0; i < Frame.DataLength; i++) { Payload[i] = Frame.Data[i]; }
+                    }
+                    return new Tuple<uint, byte[]>(Frame.CANID, Payload);
+                }
+            });
+        }
+
+        /// <summary> Converts a DLC into a proper multiple of 4 for CAN transmission </summary>
+        /// <returns> The multipole of 4 </returns>
+        /// <param name="CanDLC"> Can dlc. </param>
```

Comment:
Unless you are explaining something about the parameter, don't bother putting this. It's a waste of space.